### PR TITLE
Add support for LND 0.15.0

### PR DESCRIPTION
**Pull request by AaronDewes**
Except the minor changes in autogenerated-types, everything else was automatically generated.

This will probably break compatibility with LND 0.14.3

### CODE REVIEW

File: lnrpc/stateservice.ts [hunk 1]
```diff
@@ -155,24 +152,20 @@ export const SubscribeStateResponse = {
...
-    return message;
+    return {
+      state: isSet(object.state) ? walletStateFromJSON(object.state) : 0,
+    };
   },
 
   toJSON(message: SubscribeStateResponse): unknown {
-    const object: any = {};
+    const obj: any = {};
     message.state !== undefined &&
```
Comment (WilliamConnatser) [inline lnrpc/stateservice.ts:166]: TBH I really hate this syntax- I prefer a bracket-less if statement for stuff like this.

Comment (WilliamConnatser) [reply lnrpc/stateservice.ts:166]: What about this?

```
return {
  state: message.state ? walletStateToJSON(message.state) : undefined
}
```

File: lnrpc/stateservice.ts [hunk 2]
```diff
@@ -208,18 +201,16 @@ export const GetStateRequest = {
...
   fromJSON(_: any): GetStateRequest {
-    const message = createBaseGetStateRequest();
-    return message;
+    return {};
   },
 
   toJSON(_: GetStateRequest): unknown {
-    const object: any = {};
-    return object;
+    const obj: any = {};
```
Comment (WilliamConnatser) [inline lnrpc/stateservice.ts:212]: Same syntax nitpick here

File: lnrpc/stateservice.ts [hunk 3]
```diff
@@ -259,24 +249,20 @@ export const GetStateResponse = {
...
-        : 0;
-    return message;
+    return {
+      state: isSet(object.state) ? walletStateFromJSON(object.state) : 0,
+    };
   },
 
   toJSON(message: GetStateResponse): unknown {
-    const object: any = {};
+    const obj: any = {};
```
Comment (WilliamConnatser) [inline lnrpc/stateservice.ts:262]: And here

**Review by WilliamConnatser — CHANGES_REQUESTED** [review]
Just a few nitpicks. I guess the LND API changed?? I don't understand how the changes are used, or why they are necessary tbh.

File: lnrpc/stateservice.ts (hunk 3 above)
Comment (AaronDewes) [reply lnrpc/stateservice.ts:262]: This is autogenerated code, I don't want to change it.

**Comment by AaronDewes** [comment]
> Just a few nitpicks. I guess the LND API changed?? I don't understand how the changes are used, or why they are necessary tbh.

Yes, this basically updates the autogenerated code so we can more easily use LND 0.15.0 features in later versions.

**Review by WilliamConnatser — APPROVED** [review]

**Review by pwltr — APPROVED** [review]

**Pull request merged**